midi_app_controller/state/state_manager.py

```diff
@@ -150,6 +150,19 @@ def select_controller(self, controller_file: Optional[Path]) -> None:
             Controller.load_from(controller_file).name, controller_file
         )
 
+        if not Config.APP_STATE_FILE.exists():
+            return
+
+        recent_midi_ports_for_controller = AppState.load_from(
+            Config.APP_STATE_FILE
+        ).recent_midi_ports_for_controller
+        if self.selected_controller.path in recent_midi_ports_for_controller:
+            ports = recent_midi_ports_for_controller[self.selected_controller.path]
+            if ports["in"] in self.get_available_midi_in():
+                self.selected_midi_in = ports["in"]
+            if ports["out"] in self.get_available_midi_out():
+                self.selected_midi_out = ports["out"]
```

Comment:
I think that it should be loaded in `load_state`, not here. Just like `recent_binds_for_controller` are handled.

Reply:
I'm not sure. We might want to switch controllers multiple times in one session, and load_state is only for loading state at the beginning.

Reply:
The `load_state` is not necessary because the data would be stored in a dict (initialized in `load_state`, saved in `save_state`, and updated after selecting ports). Just like with `recent_binds_for_controller`.

Reply:
For example `recent_binds_for_controller` are updated in `select_binds()`. They are not saved/loaded from disk.

Reply:
@modyf01

Reply:
@modyf01